src/plugins/dashboard/public/dashboard_container/state/dashboard_container_reducers.ts

```diff
@@ -116,10 +116,18 @@ export const dashboardContainerReducers = {
     state.componentState.lastSavedInput = action.payload;
   },
 
+  /**
+   * Resets the dashboard to the last saved input, excluding:
+   * 1) The time range, unless `timeRestore` is `true` - if we include the time range on reset even when
+   *    `timeRestore` is `false`, this causes unecessary data fetches for the control group.
+   * 2) The view mode, since resetting should never impact this - sometimes the Dashboard saved objects
+   *    have this saved in and we don't want resetting to cause unexpected view mode changes.
+   */
   resetToLastSavedInput: (state: DashboardReduxState) => {
     state.explicitInput = {
       ...state.componentState.lastSavedInput,
-      viewMode: state.explicitInput.viewMode, // keep current view mode when resetting
+      ...(!state.explicitInput.timeRestore && { timeRange: state.explicitInput.timeRange }),
```

Comment:
When `timeRestore` is `false`, the `timeRange` of the Dashboard is never saved into the `lastSavedInput` (i.e. it is always `undefined`) - this means that, if we unnecessarily include it when resetting, this causes the control group input to be updated **twice**:

1) when the time range is set to `undefined`, this causes the following subscription to fire which then calls `updateInput` on the control group - note that, at this point, the selections are usually still the "new" version (i.e. the version that is causing unsaved changes):
   
https://github.com/elastic/kibana/blob/6e190499e4a31082fc6b5b76b230b6aa068a6027/src/plugins/dashboard/public/dashboard_container/embeddable/create/controls/dashboard_control_group_integration.ts#L86-L107

2) `this.controlGroup.updateInput(lastSavedControlGroupInput);` from the `resetToLastSavedState()` gets called because the selections + time range aren't equal. Note that, before this PR, this **also** happened as a result of a subscription; so, the order of these two input changes was a lot less predictable.

So, depending on the order that the `updateInputs` from (1) and (2) are executed in (especially when (1) was caused by a subscription firing, as mentioned above), you could sometimes get unexpected results.